db/crud.go

```diff
@@ -1899,7 +1899,8 @@ type updateAndReturnDocCallback func(*Document) (resultDoc *Document, resultAtta
 //  1. Receive the updated document body in the response
 //  2. Specify the existing document body/xattr/cas, to avoid initial retrieval of the doc in cases that the current contents are already known (e.g. import).
 //     On cas failure, the document will still be reloaded from the bucket as usual.
-func (db *DatabaseCollectionWithUser) updateAndReturnDoc(ctx context.Context, docid string, allowImport bool, expiry *uint32, opts *sgbucket.MutateInOptions, existingDoc *sgbucket.BucketDocument, callback updateAndReturnDocCallback) (doc *Document, newRevID string, err error) {
+//  3. If isImport=true, document body will not be updated - only metadata xattr(s)
+func (db *DatabaseCollectionWithUser) updateAndReturnDoc(ctx context.Context, docid string, allowImport bool, expiry *uint32, opts *sgbucket.MutateInOptions, existingDoc *sgbucket.BucketDocument, isImport bool, callback updateAndReturnDocCallback) (doc *Document, newRevID string, err error) {
```

Comment:
Take or leave this comment as maybe out of scope.

I noticed that `allowImport` and `isImport` are both args to this function and I dug deeper in the code to figure out why because these variables seem very similar and it was not clear why they would both need to be set and are different. As far as I can tell, `allowImport` could be removed because https://github.com/couchbase/sync_gateway/blob/c4ba903c5d3096088a418bf440ee2b73b8321938/db/crud.go#L1809 could say:

```
err = validateExistingDoc(doc, col.UseXattrs(), docExists)
```

and this would avoid the need to pass in `allowImport` at all.

Reply:
Understood, I was commenting on the fact that these have very similar argument names and it was a bit confusing to me.

It is different than this PR, only related in that I look at the function signature and saw two bool arguments with import in the name and wondered how they were different.